src/main/kotlin/me/fornever/todosaurus/issueTrackers/ui/wizard/ChooseIssueTrackerStep.kt

```diff
@@ -0,0 +1,319 @@
+import com.intellij.openapi.project.Project
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.JBColor
+import com.intellij.ui.UserActivityWatcher
+import com.intellij.ui.components.JBCheckBox
+import com.intellij.ui.dsl.builder.Align
+import com.intellij.ui.dsl.builder.panel
+import com.intellij.ui.layout.ComponentPredicate
+import kotlinx.coroutines.*
+import me.fornever.todosaurus.TodosaurusBundle
+import me.fornever.todosaurus.issueTrackers.*
+import me.fornever.todosaurus.issueTrackers.anonymous.AnonymousCredentials
+import me.fornever.todosaurus.issueTrackers.ui.controls.IssueTrackerComboBox
+import me.fornever.todosaurus.issueTrackers.ui.controls.IssueTrackerCredentialsComboBox
+import me.fornever.todosaurus.issueTrackers.ui.controls.ServerHostComboBox
+import me.fornever.todosaurus.ui.wizard.DynamicStepProvider
+import me.fornever.todosaurus.ui.wizard.MemorableStep
+import me.fornever.todosaurus.ui.wizard.TodosaurusContext
+import me.fornever.todosaurus.ui.wizard.TodosaurusStep
+import javax.swing.JButton
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+class ChooseIssueTrackerStep(private val project: Project, private val model: TodosaurusContext)
+    : TodosaurusStep(), DynamicStepProvider, MemorableStep {
+    companion object {
+        val id: Any = ChooseIssueTrackerStep::class.java
+    }
+
+    override val id: Any = Companion.id
+
+    private var issueTrackerPicker: IssueTrackerComboBox = IssueTrackerComboBox()
+    private var serverHostPicker: ServerHostComboBox = ServerHostComboBox()
+    private var credentialsPicker: IssueTrackerCredentialsComboBox = IssueTrackerCredentialsComboBox()
+    private var connectAnonymouslyCheckBox: JBCheckBox = JBCheckBox(TodosaurusBundle.message("wizard.steps.chooseIssueTracker.account.connectAnonymously"))
+    private var testConnectionResultLabel: JLabel = JLabel()
+    private lateinit var testConnectionButton: JButton
+
+    private var cachedServerPaths: Map<String, ServerPath>? = null
+    private var cachedCredentials: Map<String, List<IssueTrackerCredentials>?>? = null
+
+    init {
+        issueTrackerPicker.removeAllItems()
+
+        credentialsPicker.addItemListener {
+            testConnectionResultLabel.text = ""
+        }
+
+        serverHostPicker.addItemListener { event ->
+            val serverHost = event.item as? String ?: return@addItemListener
+
+            credentialsPicker.removeAllItems()
+            testConnectionResultLabel.text = ""
+
+            cachedCredentials
+                ?.getValue(serverHost)
+                ?.forEach {
+                    credentialsPicker.addItem(it)
+                }
+        }
+
+        issueTrackerPicker.addItemListener { event ->
+            val issueTracker = event.item as? IssueTracker ?: return@addItemListener
+
+            serverHostPicker.removeAllItems()
+            testConnectionResultLabel.text = ""
+
+            val credentials = runBlocking {
```

Comment:
`runBlocking` in most cases is a very bad idea: it either deadlocks the UI or causes various UX problems.

We should consider either introducing the "loading" state for the underlying controls and make this spin in background, or at lease add a modal cancellable progress bar here.

Feel free to leave a TODO here if you don't want to deal with it right now, though.